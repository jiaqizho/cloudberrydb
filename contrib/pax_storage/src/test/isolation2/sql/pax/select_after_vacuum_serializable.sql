-- @Description Ensures that a serializable select before during a vacuum operation blocks the vacuum.
-- 
DROP TABLE IF EXISTS pax_tbl;
DROP TABLE IF EXISTS pax_tbl2;
CREATE TABLE pax_tbl (a INT);
CREATE TABLE pax_tbl2 (a INT);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl select generate_series(1,1000);
insert into pax_tbl2 select generate_series(1,1000);

DELETE FROM pax_tbl WHERE a < 128;
1: BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE;
1: SELECT COUNT(*) FROM pax_tbl2;
2: VACUUM pax_tbl;
1: SELECT COUNT(*) FROM pax_tbl;
1: COMMIT;
3: INSERT INTO pax_tbl VALUES (0);
