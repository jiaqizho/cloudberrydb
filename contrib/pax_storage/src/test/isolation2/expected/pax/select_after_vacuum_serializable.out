-- @Description Ensures that a serializable select before during a vacuum operation blocks the vacuum.
--
DROP TABLE IF EXISTS pax_tbl;
DROP
DROP TABLE IF EXISTS pax_tbl2;
DROP
CREATE TABLE pax_tbl (a INT);
CREATE
CREATE TABLE pax_tbl2 (a INT);
CREATE
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl select generate_series(1,1000);
INSERT 1000
insert into pax_tbl2 select generate_series(1,1000);
INSERT 1000

DELETE FROM pax_tbl WHERE a < 128;
DELETE 2667
1: BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE;
BEGIN
1: SELECT COUNT(*) FROM pax_tbl2;
 count 
-------
 1000  
(1 row)
2: VACUUM pax_tbl;
VACUUM
1: SELECT COUNT(*) FROM pax_tbl;
 count 
-------
 18333 
(1 row)
1: COMMIT;
COMMIT
3: INSERT INTO pax_tbl VALUES (0);
INSERT 1
